tests/social_smells_tests.R

```diff
@@ -0,0 +1,75 @@
+library("testthat")
+library("kaiaulu")
+library("data.table")
+library("stringi")
+source("~/kaiaulu/R/interval.R")
+
+commit_hash_one <- "319857233b75121f357801cbdd9b5028e3057dbd"
+commit_hash_two <- "83c52a8d7ab0be9215a70351b2a4d27938092d72"
+commit_hash_three <- "9e7ba8cbacd5d09eb2af4c76eb82f7df5e0b8739"
+
+datetime_one <- "2006-04-23 11:16:01"
+datetime_two <- "2006-6-11 20:12:42"
+datetime_three <- "2006-05-24 21:45:31"
+
+pathname_one <- "path/to/file/one.c"
+pathname_two <- "path/to/file/two.py"
+pathname_three <- "path/to/file/three.py"
+
+commit_hash <- c(commit_hash_one, commit_hash_two, commit_hash_three)
+author_datetimetz <- c(datetime_one, datetime_two, datetime_three)
+file_pathname <- c(pathname_one, pathname_two, pathname_three)
+
+test_that("Datetime found amongst multiple rows", {
+  minimal_dt = data.table(author_datetimetz, commit_hash, file_pathname)
+
+  output_datetime <- get_date_from_commit_hash(minimal_dt, commit_hash_one)
+  expect_equal(output_datetime, datetime_one)
+})
+
+test_that("Null returned when table doesn't have author_datetimetz column", {
+  minimal_dt = data.table("random_name" = author_datetimetz, commit_hash, file_pathname)
+  output_datetime <- get_date_from_commit_hash(minimal_dt, commit_hash_one)
+  expect_equal(output_datetime, NULL)
+})
+
+# test_that("NA returned when table doesn't have commit_hash column", {
+#   minimal_dt = data.table(author_datetimetz, "random_name" = commit_hash, file_pathname)
+#   output_datetime <- get_date_from_commit_hash(minimal_dt, commit_hash_to_find)
+#   expect_equal(is.na(output_datetime), TRUE)
+# })
```

Comment:
This works as intended for me, i.e. it returns NA. 

I noticed your "commit_hash_to_find" is not defined at this point in the code. Maybe you had it pre-loaded with a commit hash in minimal_dt and forgot to delete it?

R is pretty sloppy with variable scope, so you may end up seeing variables not defined in a function working if they are defined outside, even if not passed as parameter. Can you try to re-run this by defining `commit_hash_to_fnd` in this function too?

Reply:
Interesting, I'm working with it currently and I changed the `commit_hash_to_find` to `commit_hash_one` then cleaned up my environment variables before running again. It still returns the corresponding datetime for the inputted commit_hash. I then replace it with `commit_hash_two` and then get the corresponding datetime for that as well.